src/test/java/io/jenkins/plugins/pipelinegraphview/cards/items/UpstreamCauseRunDetailsItemTest.java

```diff
@@ -0,0 +1,67 @@
+package io.jenkins.plugins.pipelinegraphview.cards.items;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Result;
+import io.jenkins.plugins.pipelinegraphview.cards.RunDetailsItem;
+import io.jenkins.plugins.pipelinegraphview.utils.TestUtils;
+import java.util.Optional;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+@WithJenkins
+class UpstreamCauseRunDetailsItemTest {
```

Comment:
I worry that these tests are rather heavy for what they are checking but I didn't know how to easily create a workflow run and all the other surrounding infrastructure required.